app/controllers/question_answer/comments_controller.rb

```diff
@@ -64,6 +75,12 @@ def comments_params
       params.permit(:post_id, :last_comment_id, :raw)
     end
 
+    def find_comment(comment_id)
+      comment = QuestionAnswerComment.find_by(id: comment_id)
+      raise Discourse::NotFound if comment.blank?
```

Comment:
I guess this happens if the user is editing an older version of the page? Btw what's the UX when we raise the error?

Reply:
> I guess this happens if the user is editing an older version of the page?

This can happen if we somehow use the API incorrectly on the client side as well. When that happens we want to return the right response. 

> Btw what's the UX when we raise the error?

This is unlikely to happen but we popup the error using a modal on the client side.